featurebyte/session/base.py

```diff
@@ -490,7 +559,12 @@ async def get_async_query_stream(
         try:
             # execute in separate thread
             await to_thread(
-                cursor.execute, timeout, query, **self.get_additional_execute_query_kwargs()
+                self._execute_query,
+                timeout,
+                lambda cursor, query, **kwargs: self._try_cancel_query(cursor, query),
```

Comment:
Does this lambda need to have `cursor` and `query` as the arguments or can it simply be:

```
lambda: self._try_cancel_query(cursor, query)
```

Reply:
Indeed. That should work